connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/CustomSparkConnectBlockingStub.scala

```diff
@@ -18,13 +18,93 @@ package org.apache.spark.sql.connect.client
 
 import scala.jdk.CollectionConverters._
 
+import com.google.protobuf.GeneratedMessageV3
 import io.grpc.ManagedChannel
+import io.grpc.stub.StreamObserver
 
 import org.apache.spark.connect.proto._
 
+// This is common logic shared between the blocking and non-blocking stubs.
+//
+// The common logic is responsible to verify the integrity of the response. The invariant is
+// that the same stub instance is used for all requests from the same client.
+private[client] class SparkConnectCommonStub {
+  // Server side session ID, used to detect if the server side session changed. This is set upon
+  // receiving the first response from the server. This value is used only for executions that
+  // do not use server-side streaming.
+  private var serverSideSessionId: Option[String] = None
```

Comment:
this will be different in blocking and non-blocking stub, technically possibly leading to the two operating with different ids. Should there be a common stub-state object for that?

If you make it a ResponseValidator object, it can there share this state.

Why not in general have the CustomSparkConnectCommonStub object be a container that holds instances of ResponseValidator, RetryHandler and GrpcExceptionConverter that can be shared between the blocking and non-blocking stub?

Reply:
I moved it to shared state between the stubs